lib/puppet/provider/elasticsearch_plugin/plugin.rb

```diff
@@ -0,0 +1,157 @@
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__),"..","..",".."))
+
+Puppet::Type.type(:elasticsearch_plugin).provide(:plugin) do
+  desc "A provider for the resource type `elasticsearch_plugin`,
+        which handles plugin installation"
+
+  commands :plugin => '/usr/share/elasticsearch/bin/plugin'
+  commands :es => '/usr/share/elasticsearch/bin/elasticsearch'
+
+  def exists?
+    es_version
+    if !File.exists?(pluginfile)
+      debug "Plugin file #{pluginfile} does not exist"
+      return false
+    elsif File.exists?(pluginfile) && readpluginfile != pluginfile_content
+      debug "Got #{readpluginfile} Expected #{pluginfile_content}"
+      self.destroy
+      false
+    else
+      debug "Plugin exists"
+      return true
+    end
+  end
```

Comment:
I wanted to have the debug statements in there at all parts so its easier to trace down issues when running with --debug flag.